### ISSUE TITLE
Port Checker

### ISSUE THREAD

**Issue by artyPD** [CorentinTh/it-tools#1275]
### What type of request is this?

New tool idea

### Clear and concise description of the feature you are proposing

A tool to find out what ports are open for a specified host (IP or domain or FQDN).

## [Port Checker](https://dnschecker.org/port-scanner.php)
- Ideally be able to check multiple specified ports like this:

![image](https://github.com/user-attachments/assets/11995125-8cb7-4cfc-bea0-f3c1c00d178e)
![image](https://github.com/user-attachments/assets/89dd4363-b78d-4dec-aa1d-d03c185de20c)

### Existing request
A quick search through the issues showed no previous request matching this.

### Client side only
I assume this feature would work since the client would be able to connect to the specified host.

### Validations

- [X] Check the feature is not already implemented in the project.
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Check that the feature can be implemented in a client side only app (IT-Tools is client side only, no server).

**Comment by sharevb** [comment]
Hi @artyPD, this is and will never be possible on client side (in browser). This require a server side or a real terminal. So not possible in this project

**Comment by artyPD** [comment]
Hi @sharevb,

Why is that? Is it not possible for the client to make requests to remote hosts in the browser?

**Comment by sharevb** [comment]
Hi @artyPD, javascript ran in browser can only make HTTP(s) requests and due to CORS, to only allowed targets ; or websocket. It cannot ping (ICMP) or connect to any random. Else imagine what hackers could do with anybodies computers.

**Comment by artyPD** [comment]
I see. I did some reading up and found that browsers are unable to make use of the Network layer (Layer 3) of the OSI model. Rather, browsers operate at the Application layer (Layer 7) and the Transport layer (Layer 4) by design. Also, as you mentioned, due to CORS and the security risks involved browser are therefore unable to make ICMP requests.

Thanks @sharevb. Your response is much appreciated. Closing issue now.
